### ISSUE TITLE
Parallel properties

### ISSUE THREAD

**Issue by jjl** [whatyouhide/stream_data#106]
It would be nice to be able to run a single property in parallel in order that the full run might finish quicker.

I have a complex property that starts to take a very long time towards the end of the run (diagnosed by adding `IO.inspect(:ok)` to the end of each run). I'm able to throw cores at the problem, but currently, StreamData isn't capable of using them.

Is there interest in this? Would a contribution to add this be welcome?

**Comment by luc-tielen** [comment]
This would be a cool feature imo.

**Comment by whatyouhide** [comment]
@jjl yes, this is a good way of improving StreamData. It was proposed before to do this by parallelizing properties (instead of inside a single property), but that would require major changes to ExUnit, so I would consider this solution better. @jjl would you have time to experiment with this and send out a proof of concept PR? We should likely use `Task.async_stream` for this, and start shrinking on the first error that we get (basically pipe into `Enum.find` or something like that to stop the stream). Let me know if you would like to work on this.

**Comment by jjl** [comment]
I'll see what I can do about it, but time is a little constrained at the minute, so more likely to be later than now.

**Comment by jjl** [comment]
I think the easiest way would be to make `check all` wrap the body in an extra function before passing to `gen all` and then do a `Task.async_stream(fns, fn x -> x.())`.

Does this sound like a reasonable approach?

**Comment by whatyouhide** [comment]
@jjl yes, that sounds about right. :)

**Comment by whatyouhide** [comment]
Closing in favour of the PR (#112).
